lib/src/lib.rs

```diff
@@ -217,6 +217,25 @@ impl Request {
     }
 }
 
+impl Serialize for Request {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut s = s.serialize_struct("request", 4)?;
```

Comment:
```suggestion
        let mut s = s.serialize_struct("Request", 4)?;
```

Reply:
Also, the `4` isn't always correct, is it? I am not sure what implications it has but I think we need to make this dynamic based on whether we will be serializing `params` or not.